homeassistant/components/triggercmd/__init__.py

```diff
@@ -0,0 +1,30 @@
+"""The TRIGGERcmd component."""
+
+from __future__ import annotations
+
+from triggercmd import ha
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import TOKEN
+
+PLATFORMS = [
+    Platform.SWITCH,
+]
+
+type TriggercmdConfigEntry = ConfigEntry[ha.Hub]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: TriggercmdConfigEntry) -> bool:
+    """Set up TRIGGERcmd from a config entry."""
+    entry.runtime_data = ha.Hub(entry.data[TOKEN])
```

Comment:
Can we double check that everything works here?

Reply:
I assume this is what you mean, but let me know if I misunderstood:
https://developers.home-assistant.io/docs/integration_setup_failures#integrations-using-async_setup_entry